File: frontend/lib/weekdayPattern.ts
import { apiFetch } from "@/lib/api";

export type WeekdayPoint = {
  weekday: number;
  label: string;
  avg_quantity: number;
};

export type WeekdayPatternResponse = {
  product_id: number;
  window_days: number;
  points: WeekdayPoint[];
};

export async function fetchWeekdayPattern(
  productId: number,
  windowDays = 90
) {
  const params = new URLSearchParams({
    window_days: String(windowDays),
  });
  return apiFetch<WeekdayPatternResponse>(
    `/api/products/${productId}/weekday-pattern?${params.toString()}`
  );
}

File: app/schemas/weekday_pattern.py
from typing import List

from pydantic import BaseModel


class WeekdayValue(BaseModel):
    weekday: int  # 0 = Monday
    label: str
    avg_quantity: float


class WeekdayPatternResponse(BaseModel):
    product_id: int
    window_days: int
    points: List[WeekdayValue]




